test/e2e/all_in_one_test.go

```diff
@@ -230,6 +236,115 @@ func TestDeployAllInOnePostgres(t *testing.T) {
+
+	t.Log("deploying kong components")
+	manifest, err := getTestManifest(t, postgresPath)
+	require.NoError(t, err)
+	deployment := deployKong(ctx, t, env, manifest)
+
+	t.Log("this deployment used a postgres backend, verifying that postgres migrations ran properly")
+	verifyPostgres(ctx, t, env)
+
+	t.Log("running ingress tests to verify all-in-one deployed ingress controller and proxy are functional")
+	verifyIngress(ctx, t, env)
+
+	forDeployment := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("app=%s", deployment.Name),
+	}
+
+	podList, err := env.Cluster().Client().CoreV1().Pods(deployment.Namespace).List(ctx, forDeployment)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(podList.Items))
+	initialPod := podList.Items[0]
+
+	scale := &autoscalingv1.Scale{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deployment.Name,
+			Namespace: deployment.Namespace,
+		},
+		Spec: autoscalingv1.ScaleSpec{
+			Replicas: 2,
+		},
+	}
+	t.Log("adding a second replica to the Kong deployment")
+	_, err = env.Cluster().Client().AppsV1().Deployments(deployment.Namespace).UpdateScale(ctx,
+		deployment.Name, scale, metav1.UpdateOptions{})
+	require.NoError(t, err)
+
+	require.Eventually(t, func() bool {
+		deployment, err = env.Cluster().Client().AppsV1().Deployments(deployment.Namespace).Get(ctx, deployment.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return deployment.Status.ReadyReplicas == *deployment.Spec.Replicas
+	}, kongComponentWait, time.Second)
+
+	podList, err = env.Cluster().Client().CoreV1().Pods(deployment.Namespace).List(ctx, forDeployment)
+	require.NoError(t, err)
+	var secondary corev1.Pod
+	for _, pod := range podList.Items {
+		if pod.Name != initialPod.Name {
+			secondary = pod
+			break
+		}
+	}
+
+	t.Log("confirming the second replica is not the leader and is not pushing configuration")
+	forwardCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	startPortForwarder(forwardCtx, t, env, secondary, "9777", "cmetrics")
+	require.Never(t, func() bool {
+		req, err := http.NewRequest("GET", "http://localhost:9777/metrics", nil)
+		require.NoError(t, err)
+		client := &http.Client{}
+		resp, err := client.Do(req)
+		require.NoError(t, err)
+		defer resp.Body.Close()
+		body, err := io.ReadAll(resp.Body)
+		require.NoError(t, err)
+		// if we are not the leader, we run no config pushes, and this metric string will not appear
+		return strings.Contains(string(body), metrics.MetricNameConfigPushCount)
+	}, time.Minute, time.Second*10)
+
+	t.Log("deleting the original replica and current leader")
+	err = env.Cluster().Client().CoreV1().Pods(initialPod.Namespace).Delete(ctx, initialPod.Name, metav1.DeleteOptions{})
+	require.NoError(t, err)
+
+	t.Log("confirming the second replica becomes the leader and starts pushing configuration")
+	require.Eventually(t, func() bool {
+		req, err := http.NewRequest("GET", "http://localhost:9777/metrics", nil)
+		require.NoError(t, err)
+		client := &http.Client{}
```

Comment:
Since we're also using a `net/http` client here, probably should just make a default `httpc` as suggested before and use it across both.

Reply:
dc6a91039b7899a03ae49a8b2cda99483e0ce090